packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SubjectGroupViewer/SubjectGroupViewerContainer.js

```diff
@@ -138,7 +138,7 @@ class SubjectGroupViewerContainer extends React.Component {
     if (subject && subject.locations) {
       // TODO: Validate for allowed image media mime types
       
-      const imageUrls = subject.locations.map(obj => Object.values(obj)[0])
+      let imageUrls = subject.locations.map(obj => Object.values(obj)[0])
```

Comment:
We do this so often in our code (getting URLs from `subject.locations`) that we should probably write a helper for it at some point. Or write a view in the subject model that does it for us.

Reply:
We should be validating the location here to check that it actually is an image. I've written helpers for this, check out the `DataImageViewerContainer` component, line 49 using `findLocationsByMediaType`.

Reply:
I just got this down to...

```
const locations = findLocationsByMediaType(subject.locations, 'image') || []
const images = await Promise.all(
  locations.map(loc => this.fetchImage(Object.values(loc)[0]))
)
```

⚠️ ❗  ...but this made me realise it could lead to another problem. We're implicitly expecting the 'group' Subject resource from the backend to have exactly (say) 25 _image_ locations and a matching 25 _subject IDs._ (25 is an arbitrary number for illustrative purposes) If there is a mismatch in this - e.g. half the media locations are for videos/JSONs/txts - then we'd have 18 image and 25 subject IDs, causing a _complete mismatch in the image-subjectID pairings._

I'm going to need to think of a better error handling for incorrect/unexpected 'group' Subject Resources. For now, I'll keep the older doesn't-verify-for-media-type implementation since an incorrect media type just straight up **crashes** the Classifier, which is preferable to incorrect classifications, IMO.